plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/embeddings/callables/FullNamedFunctionSignature.kt

```diff
@@ -17,6 +20,26 @@ interface FullNamedFunctionSignature : NamedFunctionSignature {
     fun getPreconditions(returnVariable: VariableEmbedding): List<ExpEmbedding>
     fun getPostconditions(returnVariable: VariableEmbedding): List<ExpEmbedding>
     val declarationSource: KtSourceElement?
+    val isPrimaryConstructor: Boolean
+
+    private fun primaryConstructorFieldsWithParams(): List<Pair<FieldEmbedding, VariableEmbedding>> {
+        if (!isPrimaryConstructor) return emptyList()
+        return (returnType as? ClassTypeEmbedding)?.fields?.values?.mapNotNull { field ->
+            val correspondingParameterSymbol = field.symbol?.correspondingValueParameterFromPrimaryConstructor
+            val correspondingParameter = symbolsToParams[correspondingParameterSymbol]
+            if (correspondingParameter == null) null
+            else field to correspondingParameter
```

Comment:
I think this can be simplified to something like `symbolsToParams[correspondingParameterSymbol]?.let { field to it }`